clients/admin-ui/src/features/privacy-requests/privacy-requests.slice.ts

```diff
@@ -440,10 +460,26 @@ export const {
   useCreateStorageMutation,
   useCreateStorageSecretsMutation,
   useCreateConfigurationSettingsMutation,
+  useGetConfigurationSettingsQuery,
   useGetMessagingConfigurationDetailsQuery,
   useGetActiveMessagingProviderQuery,
   useGetActiveStorageQuery,
   useCreateMessagingConfigurationMutation,
   useCreateMessagingConfigurationSecretsMutation,
   useCreateTestConnectionMessageMutation,
 } = privacyRequestApi;
+
+const EMPTY_CORS_DOMAINS = { domains: [] } as { domains: string[] };
+export const selectCORSDomains = () =>
+  createSelector(
+    [
+      (state) => state,
+      privacyRequestApi.endpoints.getConfigurationSettings.select(),
+    ],
+    (state, { data }) => {
+      // the type checker was having a really hard time with `data`
+      const properlyTypedData = data as ApplicationConfig;
+      const domains = properlyTypedData?.security?.cors_origins;
```

Comment:
hmm I suppose it's because `getConfigurationSettings` has [a lot of things it could be](https://github.com/ethyca/fides/blob/cors-domain-config-api/clients/admin-ui/src/features/privacy-requests/privacy-requests.slice.ts/#L358-L362), and the backend only types it as [`Dict[str, Any]`](https://github.com/ethyca/fides/blob/cors-domain-config-api/src/fides/api/api/v1/endpoints/config_endpoints.py#L28)

the proper thing to do would likely be to set the return type of `getConfigurationSettings` to `Record<string, any>`, and then use something like the `narrow` lib or some other type predicate function to make sure the config has `security` and `cors_origins` (and so we know it's an `ApplicationConfig` type). that said, I don't think it provides all that much more safety than what you already have, so I won't be a stickler about it 😄

Reply:
oh great, I'm glad! looks great 😎